client/app/services/session.service.js

```diff
@@ -73,5 +74,69 @@
       // may not be current, but if we have one, we'll rely on API 401ing if it's not
       return model.token;
     }
+
+    function setRBAC(productFeatures) {
+      setRBACForActions(productFeatures);
+      setRBACForNavigation(productFeatures);
+    }
+
+    function setRBACForNavigation(productFeatures) {
+      var features = {
+        dashboard: {show: entitledForDashboard(productFeatures)},
+        services: {show: entitledForServices(productFeatures)},
+        requests: {show: entitledForRequests(productFeatures)},
+        marketplace: {show: entitledForServiceCatalogs(productFeatures)}
+      };
+      model.navFeatures = features;
+
+      return model.navFeatures;
+    }
+
+    function setRBACForActions(productFeatures) {
+      var features = {
+        service_view: {show: angular.isDefined(productFeatures.service_view)},
+        service_edit: {show: angular.isDefined(productFeatures.service_edit)},
+        service_delete: {show: angular.isDefined(productFeatures.service_delete)},
+        service_reconfigure: {show: angular.isDefined(productFeatures.service_reconfigure)},
+        service_retire_now: {show: angular.isDefined(productFeatures.service_retire_now)}
+      };
+      model.actionFeatures = features;
+
+      return model.actionFeatures;
+    }
+
+    function entitledForServices(productFeatures) {
+      var serviceFeature = lodash.find(model.actionFeatures, function(o) {
+        return o.show === true;
+      });
+
+      return angular.isDefined(serviceFeature);
+    }
+
+    function entitledForServiceCatalogs(productFeatures) {
+      if (angular.isDefined(productFeatures.svc_catalog_provision)) {
+        return true;
+      } else {
+        return false;
+      }
```

Comment:
Do you need the whole if/then construct?  Can't this just be...

```js
function entitledForServiceCatalogs(productFeatures) {
  return angular.isDefined(productFeatures.svc_catalog_provision);
}
```

Same goes for the other similar methods below.